Modify S2083&S6096(Education): Add Partial Path Traversal to pitfalls (#1243)
SonarSource/rspec: rules/S2083/common/pitfalls/partial-path-traversal.adoc

==== Partial Path Traversal

> When checking untrusted folders by comparing them to trusted folder names,
> **ensure the validation strings all contain a path separator as the last
> character**. +

Author: ```suggestion
When validating untrusted paths by checking if they start with trusted path names,
**ensure the validation strings all contain a path separator as the last
character**. +
```

> When checking untrusted folders by comparing them to trusted folder names,
> **ensure the validation strings all contain a path separator as the last
> character**. +
> A partial path traversal vulnerability can be unintentionally introduced into
> the application without a path separator as the last character of the
> validation strings.

Reviewer: ```suggestion
When validating untrusted paths by comparing them to trusted path names,
**ensure the validator string contains a path separator as the last character**. + 
A partial path traversal vulnerability can be unintentionally introduced into the
application due to misuse of the validator method.

```

Author: About your suggestion:
* I used the word "folders" because I realized that this partial path traversal attack was only relevant when the validation path was a folder. If it was something else, the validation would have been different
* I think that "a path separator as the last character of the validation strings." (initial text) is clearer than "due to misuse of the validator method." because it clearly states how the vulnerability can be introduced

Author: I added your suggestion here: https://github.com/SonarSource/rspec/pull/1243/files#r970864019
